### DISCUSSION TITLE
"services.db.ports contains an invalid type, it should be a number, or an object" error

### DISCUSSION THREAD

**Discussion by shodanx2** [nezhar/wordpress-docker-compose#84] [Q&A]
Hi,

Trying this container the for first time.
Downloaded the docker-compose.yml file 
Downloaded the env.example file and renamed to .env in the same folder
I edited the .env file, I put my hostname for the IP, not sure if that's ok 
and got this error

```
root@www:/opt/wordpress# docker-compose up -d
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.db.ports contains an invalid type, it should be a number, or an object
services.pma.ports contains an invalid type, it should be a number, or an object
services.wp.ports contains an invalid type, it should be a number, or an object
```

Looking in the docker-compose file I see

```
services:
  wp:
    ports:
      - ${IP}:80:80 # change ip if required
```

I suspect that ${IP} is still empty here ? 
How does that .env file work ? It doesn't seem to get invoked in the docker-compose file ?

**Comment by RenatoSalles007** [comment]
Please post your .env file

**Reply by abatista667** [reply]
the default
